ui/plugin-system/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx

```diff
@@ -127,6 +150,9 @@ export function VariableEditForm(props: VariableEditFormProps) {
               label="Name"
               value={state.name}
               helperText={validation.name}
+              InputProps={{
+                readOnly: action === 'update' || action === 'read',
```

Comment:
Maybe you can put a comment here to explain why it's disabled in update mode (aka requires a specific logic because it would make the update/PUT call to the API fail in "document not found").

Still this is probably something we want to permit at some point, and would require a DELETE + POST call then if I'm not mistaken. Maybe a quick win for now could be to have a Rename button outside of the form - aka in the list view just like we have for dashboards?

Reply:
ERRATUM it actually doesn't exist for dashboards, only the display name gets changed. Still the "technical" name for the variables is the one users will manipulate in their queries etc, so contrary to dashboards I think it's quite important to be able to change it.

Reply:
Btw I have ofc the same problematic on the datasources side, and I was thinking to do something like this, WDYT?

```typescript
const handleDatasourceUpdate = useCallback(
    (datasource: Datasource) => {
      if (datasource.metadata.name != targetedDatasource?.metadata.name) {
        // DELETE + POST logic
      } else {
        updateDatasourceMutation.mutate(datasource, {
          onSuccess: (updatedDatasource: Datasource) => {
            successSnackbar(`Datasource ${datasourceDisplayName(updatedDatasource)} has been successfully updated`);
            setEditDatasourceFormStateOpened(false);
          },
          onError: (err) => {
            exceptionSnackbar(err);
            throw err;
          },
        });
      }
    },
    [exceptionSnackbar, successSnackbar, updateDatasourceMutation]
  );
```

Reply:
DELETE then CREATE (or CREATE then DELETE) is tricky, what do you do if the second request fail :/ The UI will be desync too

Reply:
Yes in that case we'd need a proper way to ensure the 2 requests are bundled together (to have either both succeeding or none). I know SQL offers such capability but since we dont necessary have a SQL database here...

Anyway imho it's quite important to be able to do this (I mean maybe not to be tackled in this PR but well, before the v1 would be a must). From what I could think of & what we discussed live, these are the solutions I'm thinking of (ranked from easiest-but-dirtiest to hardest-but-cleanest):
- CREATE & DELETE calls, in separated transactions.
- Add a "duplicate" action, which allow you to replicate an existing var, just changing its name. Then you're free to delete the original one.
- CREATE & DELETE calls, grouped in a same transaction.
- Add a new ID attribute to the datamodel, and make it the primary key, auto managed/incremented by the backend. @Nexucis I'm lacking context (or memories...) here on why such ID/UID attribute was not put in the data model of the various Perses objects - this would have been a clean way to avoid this tricky case for sure ^^.

Reply:
The ability to edit the name of a resource is a long lasting topic :smile: 
Any solution we choose should be well thought and applied to all resource. 

To me having an UID will decrease the readability on the URL for example. Except if we do something a bit like grafana with ``<url>/projects/<myproject>/<resourceKind>/<resourceUID>/<resourceName>`` with ``resourceName`` just here for readability and not interpreted. 
But if you do so, you will want as well to be able to edit the name of the project and again same problem happen.